remix-lib/src/execution/logsManager.js

```diff
@@ -97,6 +106,40 @@ class LogsManager {
     delete this.subscriptions[subscriptionId]
   }
 
+  newFilter (filterType, params) {
+    let filterId = '0x' + crypto.randomBytes(16).toString('hex')
+    if (filterType === 'block' || filterType === 'pendingTransactions') {
+      this.filters[filterId] = { filterType }
+    }
+    if (filterType === 'filter') {
+      this.filters[filterId] = { filterType, params }
+    }
+    this.filterTracking[filterId] = {}
+    return filterId
+  }
+
+  uninstallFilter (filterId) {
+    delete this.filters[filterId]
+  }
+
+  getLogsForFilter (filterId, logsOnly) {
+    const {filterType, params} = this.filter[filterId]
+    const tracking = this.filterTracking[filterId]
+
+    if (logsOnly || filterType === 'filter') {
+      return this.getLogsFor(params || {topics: []})
+    }
+    if (filterType === 'block') {
+      let blocks = this.oldLogs.filter(x => x.type === 'block').filter(x => tracking.block === undefined || x.blockNumber >= tracking.block)
```

Comment:
What happen if `x` is null instead of undefined ? If there is no difference prefer `tracking.block` or `!!tracking.block`

Reply:
this wouldn't happen unless oldLogs is a list with undefined and in that case there is something very wrong somewhere else and it should fail here anyway